eval/stack.go

```diff
@@ -9,24 +9,25 @@ import (
 
 // Returns a stack array of the current stack.
 func (s *State) Stack() []string {
-	stack := make([]string, 0, s.depth) // will be -1 because no name at toplevel but... it's fine.
-	e := s.env
-	for {
-		if e == nil {
-			break
-		}
+	if s.depth <= 1 {
+		return nil
+	}
+	stack := make([]string, 0, s.depth-1)
+	for e := s.env; e != nil; e = e.StackParent() {
 		n := e.Name()
 		if n != "" {
 			stack = append(stack, e.Name())
 		}
-		e = e.StackParent()
 	}
 	log.Debugf("Stack() len %d, depth %d returning %v", len(stack), s.depth, stack)
 	return stack
 }
 
 // Creates a new error object with the given message and stack.
 func (s *State) Error(msg string) object.Error {
+	if log.LogDebug() {
+		log.LogVf("Error %q called", msg)
+	}
```

Comment:
Don't you have a log.DebugVf ?

Or the debug mode is conditioned by the code?

I would expect to have a method in log to check .logdebug and do not log if needed

Reply:
I was trying to make the compiler or runtime smarter and skip the log unless log level is debug (and yes I use verbose after which is above debug, mostly because of the blue color it gets on console)

so yes that log.LogDebug() returns a boolean

https://github.com/fortio/log/blob/v1.16.0/logger.go#L566-L568

I should use log.LogVerbose() though